test/cpp/interop/istio_test_echo_server.cc

```diff
@@ -0,0 +1,359 @@
+// necessary for certain tests.
+ABSL_FLAG(std::vector<std::string>, port, std::vector<std::string>({"8080"}),
+          "HTTP/1.1 ports");
+ABSL_FLAG(std::vector<std::string>, tcp, std::vector<std::string>({"9090"}),
+          "TCP ports");
+ABSL_FLAG(std::vector<std::string>, tls, std::vector<std::string>({""}),
+          "Ports that are using TLS. These must be defined as http/grpc/tcp.");
+ABSL_FLAG(std::vector<std::string>, bind_ip, std::vector<std::string>({""}),
+          "Ports that are bound to INSTANCE_IP rather than wildcard IP.");
+ABSL_FLAG(std::vector<std::string>, bind_localhost,
+          std::vector<std::string>({""}),
+          "Ports that are bound to localhost rather than wildcard IP.");
+ABSL_FLAG(std::vector<std::string>, server_first,
+          std::vector<std::string>({""}),
+          "Ports that are server first. These must be defined as tcp.");
+ABSL_FLAG(std::vector<std::string>, xds_grpc_server,
+          std::vector<std::string>({""}),
+          "Ports that should rely on XDS configuration to serve");
+ABSL_FLAG(std::string, metrics, "", "Metrics port");
+ABSL_FLAG(std::string, uds, "", "HTTP server on unix domain socket");
+ABSL_FLAG(std::string, version, "", "Version string");
+ABSL_FLAG(std::string, cluster, "", "Cluster where this server is deployed");
+ABSL_FLAG(std::string, crt, "", "gRPC TLS server-side certificate");
+ABSL_FLAG(std::string, key, "", "gRPC TLS server-side key");
+ABSL_FLAG(std::string, istio_version, "", "Istio sidecar version");
+ABSL_FLAG(std::string, disable_alpn, "", "disable ALPN negotiation");
+
+using grpc::Channel;
+using grpc::ClientAsyncResponseReader;
+using grpc::ClientContext;
+using grpc::CompletionQueue;
+using grpc::Server;
+using grpc::ServerBuilder;
+using grpc::ServerContext;
+using grpc::Status;
+
+using proto::EchoRequest;
+using proto::EchoResponse;
+using proto::EchoTestService;
+using proto::ForwardEchoRequest;
+using proto::ForwardEchoResponse;
+
+const std::string host_key = "Host";
+const std::string request_id_field = "X-Request-Id";
+const std::string service_version_field = "ServiceVersion";
+const std::string service_port_field = "ServicePort";
+const std::string status_code_field = "StatusCode";
+const std::string url_field = "URL";
+const std::string host_field = "Host";
+const std::string hostname_field = "Hostname";
+const std::string method_field = "Method";
+const std::string response_header = "ResponseHeader";
+const std::string cluster_field = "Cluster";
+const std::string istio_version_field = "IstioVersion";
+const std::string ip_field = "IP";  // The Requester’s IP Address.
+
+class EchoTestServiceImpl : public EchoTestService::Service {
+ public:
+  explicit EchoTestServiceImpl(const std::string& hostname)
+      : hostname_(hostname) {}
+
+  Status Echo(ServerContext* context, const EchoRequest* request,
+              EchoResponse* response) override {
+    std::stringstream ss;
+
```

Comment:
Overall, the C++ style guide https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace advocates minimal use of blank lines within functions.

Reply:
Deleted most of the vertical whitespaces.